packages/payment-detection/src/provider.ts

```diff
@@ -1,21 +1,34 @@
-import { ethers } from 'ethers';
+import { ethers, providers } from 'ethers';
 
-let configuration = {
+let defaultProviderOptions: Record<string, string> = {
   // fallback to Ethers v4 default projectId
   infura: process.env.RN_INFURA_KEY || '7d0d81d0919f4f05b9ab6634be01ee73',
 };
 
-export const initPaymentDetectionProvider = (config: Partial<typeof configuration>) => {
-  configuration = { ...configuration, ...config };
+let blockchainRpcs: Record<string, string> = {
+  matic: 'https://rpc-mainnet.matic.network/',
 };
 
-export const getDefaultProvider = (network?: string | ethers.providers.Network | undefined) => {
-  if (network === 'private') {
-    return new ethers.providers.JsonRpcProvider();
-  }
-  if (configuration.infura) {
-    return new ethers.providers.InfuraProvider(network, process.env.RN_INFURA_KEY);
+export const initPaymentDetectionProvider = (overrides: {
+  /** */
+  blockchainRpcs?: Record<string, string>;
+  /** Options as specified in https://docs.ethers.io/v5/api/providers/#providers-getDefaultProvider  */
+  defaultProviderOptions?: Record<string, string>;
+}): void => {
+  blockchainRpcs = { ...blockchainRpcs, ...overrides.blockchainRpcs };
+  defaultProviderOptions = { ...defaultProviderOptions, ...overrides.defaultProviderOptions };
+};
+
+export const getDefaultProvider = (
+  network?: string | ethers.providers.Network | undefined,
+  options?: Record<string, string>,
+): providers.Provider => {
+  if (typeof network === 'string') {
+    const rpc = blockchainRpcs[network];
+    if (rpc) {
+      return new providers.JsonRpcProvider(rpc);
+    }
   }
 
-  return ethers.getDefaultProvider(network);
+  return ethers.getDefaultProvider(network, { ...defaultProviderOptions, ...options });
```

Comment:
Default of default can be hard to understand for new comers.
But I am personally comfortable with that.

Reply:
you mean passing a param to `getDefaultProvider`? it's the way `ethers` does it. 

This wrapper enables NOT passing the param every time you call `getNetworkProvider`, but instead sharing the config